extension-standard/src/main/java/com/knowre/android/extension/standard/redux/state/ReduceOnce.kt

```diff
@@ -0,0 +1,23 @@
+package com.knowre.android.extension.standard.redux.state
+
+typealias ReduceOnce<T> = ReducibleValue<ReduceOnceValue<T>>
+
+fun <T> ReduceOnce(next: T): ReduceOnce<T> =
+    ReducibleValue(ReduceOnceValue(next))
+
+@JvmInline
+value class ReduceOnceValue<out T>
+@PublishedApi internal constructor(val value: T) {
```

Comment:
조금 생소한 어노테이션이였는데 제가 의도를 제대로 이해했는지 알고 싶습니다

1. ReduceOnceValue를 모듈 외부에서 생성하는 일이 없도록 constructor의 접근제한은 internal로 설정
2. 하지만 internal로 설정 시 public inline 클래스에서의 사용이 제한되며 이를 해결하기 위해 @PublishedApi를 사용

혹시 다른 이유가 있다면 코멘트 부탁드릴게요

Reply:
넵 이해하신게 맞습니다. 다음은 PublishedApi 주석 전문입니다. 


When applied to a class or a member with internal visibility allows to use it from public inline functions and
makes it effectively public.

Public inline functions cannot use non-public API, since if they are inlined, those non-public API references
would violate access restrictions at a call site (https://kotlinlang.org/docs/reference/inline-functions.html#public-inline-restrictions).

To overcome this restriction an `internal` declaration can be annotated with the `@PublishedApi` annotation:
- this allows to call that declaration from public inline functions;
- the declaration becomes effectively public, and this should be considered with respect to binary compatibility maintaining.
---
stdlib 에서도 많이 쓰고 있는데요, `Result.kt` 를 보시면 다음과 같이 정의되어있습니다.

```kotlin
@JvmInline
public value class Result<out T> @PublishedApi internal constructor(
    @PublishedApi
    internal val value: Any?
) {
  // ...
  publich companion object {
    public inline fun <T> success(value: T): Result<T> = Result(value)
    // ...
  }
```